src/webview/html/messageAsHtml.js

```diff
@@ -72,13 +77,87 @@ $!${messageReactionListAsHtml(reactions, ownUser.user_id, allImageEmojiById)}
 `;
 };
 
-const widgetBody = (message: Message | Outbox) => template`
+/**
+ * Render the body of a message that has submessages.
+ *
+ * Must not be called on a message without any submessages.
+ */
+const widgetBody = (message: Message, ownUserId: UserId) => {
+  invariant(
+    message.submessages !== undefined && message.submessages.length > 0,
+    'should have submessages',
+  );
+
+  const widgetSubmessages = message.submessages
+    .filter(submessage => submessage.msg_type === 'widget')
+    .map(submessage => ({
+      ...submessage,
+      content: JSON.parse(submessage.content),
+    }));
+
+  const poll_widgets = widgetSubmessages.filter(widget => widget.content.widget_type === 'poll');
+  // "events" are submessages other than the main widget submessage - for
+  // instance, `vote`, `new_option`, and similar submessages. These are not
+  // documented, look at the PollData code if you're interested in what the
+  // possibilities are.
+  const events = widgetSubmessages.filter(widget => 'type' in widget.content);
+
+  if (poll_widgets.length === 0) {
+    return template`
 $!${message.content}
 <div class="special-message"
  ><p>Interactive message</p
  ><p>To use, open on web or desktop</p
 ></div>
 `;
+  }
+  const poll_widget = poll_widgets[0];
+
+  if (!poll_widget.content || !poll_widget.content.extra_data) {
+    // We don't expect this to happen in general, but there are some malformed
+    // messages lying around that will trigger this.
+    return template`$!${message.content}`;
```

Comment:
Tracing through to the callees of the webapp's `do_process_submessages` as mentioned at https://github.com/zulip/zulip-mobile/pull/4704#discussion_r646821062 , it looks like the webapp has a different behavior here:
```js
    extra_data: {question = "", options = []} = {},
```

So in particular if there's a poll that starts with no options, the webapp will go ahead and present it normally -- with `new_option` submessages, it can end up as a perfectly functional poll. We should do the same thing, unless we know of a reason to behave differently.

Reply:
I don't believe that's what this is checking for — if you do `/poll` in the webapp, the mobile app handles that fine. This is for a particular weird message that as far as I can tell, no official client will send anymore. From what I saw, the webapp threw an error in the console for the only message that I found that triggered this condition.

Do let me know which specific part of the webapp codebase you're looking at that you think handles this if that's not the case, though.